File: authorization/src/main/java/com/example/alexeyglushkov/authorization/Auth/AccountStore.java
package com.example.alexeyglushkov.authorization.Auth;

import android.support.annotation.Nullable;

import java.util.List;

/**
 * Created by alexeyglushkov on 31.10.15.
 */
public interface AccountStore {
    void putAccount(Account account) throws Exception;

    @Nullable
    Account getAccount(int key);
    int getAccountCount();
    int getMaxAccountId();

    List<Account> getAccounts();
    List<Account> getAccounts(int serviceType);
    Error removeAccount(int id);
    Error removeAll();
    Error getError();

    // TODO: provide async loader
    boolean isLoaded();
    void load();
}

File: authorization/src/main/java/com/example/alexeyglushkov/authorization/Auth/AccountStore.kt
package com.example.alexeyglushkov.authorization.Auth

/**
 * Created by alexeyglushkov on 31.10.15.
 */
interface AccountStore {
    @Throws(Exception::class)
    fun putAccount(account: Account?)

    @Throws(Exception::class)
    fun getAccount(key: Int): Account?

    val accountCount: Int
    val maxAccountId: Int
    val accounts: List<Account?>?
    fun getAccounts(serviceType: Int): List<Account?>?
    @Throws(Exception::class)
    fun removeAccount(id: Int)

    @Throws(Exception::class)
    fun removeAll()

    // TODO: provide async loader
    val isLoaded: Boolean

    @Throws(Exception::class)
    fun load()
}